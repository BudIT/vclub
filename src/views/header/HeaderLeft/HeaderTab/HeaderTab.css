.tab
{
  color: #33363A;
  text-decoration: none;
  text-align: center;
  font-size: 22px;
  font-family: 'Rubik', helvetica, sans-serif;

  padding: 14px 16px;
  transition: 0.2s;
  display: inline-block;

  background-color: inherit;
  cursor: pointer;

  border-style: none;
  outline: none;
}

.tabCurrent {
  composes: tab;
  text-decoration: underline;
}

.tab:hover {
  background-color: #FFAA00;
  /*text-decoration: line-through;*/
}
