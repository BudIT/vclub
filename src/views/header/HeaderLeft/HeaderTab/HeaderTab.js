import React, { PropTypes } from 'react';

import compose from 'recompose/compose';
import { withHandlers } from 'recompose';

import style from './HeaderTab.css';

const enhance = compose(
  withHandlers({
    // array of handlers
    onClick: props => () =>
      props.dispatch(props.changeRoom(props.children)),
  })
);

// changeRoom
function HeaderTab(props) {
  const {
    children,
    currentRoom,

    onClick,
  } = props;

  return (
    <li>
      <button
        onClick={onClick}
        className={children === currentRoom ? style.tabCurrent : style.tab}
      >
        {children}
      </button>
    </li>
  );
}

HeaderTab.propTypes = {
  // should be only one child
  children: PropTypes.element.isRequired,
  currentRoom: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default enhance(HeaderTab);
