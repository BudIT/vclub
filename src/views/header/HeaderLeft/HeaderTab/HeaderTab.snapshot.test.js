import React from 'react';
import renderer from 'react-test-renderer';
import HeaderTab from './HeaderTab';

const tabName = 'VIDEO';

test('<HeaderTab /> renders correctly', () => {
  const dispatchSpy = jest.fn();
  const rendered = renderer.create(
    <HeaderTab isCurrentTab dispatch={dispatchSpy}>
      {tabName}
    </HeaderTab>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
