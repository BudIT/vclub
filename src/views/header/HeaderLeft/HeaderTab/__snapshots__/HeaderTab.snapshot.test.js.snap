exports[`test <HeaderTab /> renders correctly 1`] = `
<li>
  <button
    className="tabCurrent"
    onClick={[Function]}>
    VIDEO
  </button>
</li>
`;
